packages/manganis/manganis-core/src/options.rs

```diff
@@ -17,9 +17,73 @@ use crate::{
     serde::Serialize,
     serde::Deserialize,
 )]
+#[non_exhaustive]
+pub struct AssetOptions {
```

Comment:
Changing this from an enum to a struct is somewhat breaking, but most users are probably using the variants directly like this instead of manually using the enum, so I don't expect this to break much user code:
```rust
const _: Asset = asset!("/assets/image.png", ImageAssetOptions::new().with_size(ImageSize::Manual { width: 52, height: 52 }));
```